src/object/object_primitive.c

```diff
@@ -11268,6 +11372,44 @@ mr_readval_string_internal (OR_BUF * buf, DB_VALUE * value, TP_DOMAIN * domain,
 			}
 		      return ER_FAILED;
 		    }
+		  /* Handle decompression if there was any */
+		  if (compressed_size > 0)
+		    {
+		      /* String was compressed */
+		      decompression_size = 0;
+
+		      /* Handle decompression */
+		      decompressed_string = db_private_alloc (NULL, decompressed_size + 1);
+		      if (decompressed_string == NULL)
+			{
+			  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+				  (size_t) decompressed_size * sizeof (char));
+			  return rc;
+			}
+
+		      /* decompressing the string */
+		      rc =
+			lzo1x_decompress ((lzo_bytep) new_, (lzo_uint) compressed_size, decompressed_string,
+					  &decompression_size, NULL);
+		      if (rc != LZO_E_OK)
+			{
+			  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_LZO_DECOMPRESS_FAIL, 1, 0);
+			  return ER_IO_LZO_DECOMPRESS_FAIL;
```

Comment:
@lucianlazar2211 

Please notice that msg for `ER_IO_LZO_DECOMPRESS_FAIL` has no argument. 

```C
  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_LZO_DECOMPRESS_FAIL, 0);
```